source/agora/consensus/validation/Block.d

```diff
@@ -137,13 +137,18 @@ public string isInvalidReason (const ref Block block, Height prev_height,
     if (!block.txs.isSorted())
         return "Block: Transactions are not sorted";
 
+    bool only_coinbase = true;
     foreach (const ref tx; block.txs)
     {
+        only_coinbase &= tx.type == TxType.Coinbase;
         if (auto fail_reason = VTx.isInvalidReason(tx, findUTXO,
             block.header.height, checkFee))
             return fail_reason;
     }
 
+    if (only_coinbase)
+        return "Block: Coinbase only block";
```

Comment:
```suggestion
        return "Block: Must contain other transactions than Coinbase";
```

We should always aim to tell what is wrong, not just the state. As a user, would you rather see "Error: Block has 1337 transactions" or "Error: Number of transaction per block must be a power of 2, not 1337" ?